sdks/python/apache_beam/testing/test_stream_test.py

```diff
@@ -245,6 +247,82 @@ def fired_elements(elem):
     # TODO(BEAM-3377): Remove after assert_that in streaming is fixed.
     self.assertEqual([('k', ['a'])], result)
 
+  def test_basic_execution_sideinputs_batch(self):
+
+    # TODO(BEAM-3377): Remove after assert_that in streaming is fixed.
+    global result     # pylint: disable=global-variable-undefined
+    result = []
+
+    def recorded_elements(elem):
+      result.append(elem)
+      return elem
+
+    options = PipelineOptions()
+    options.view_as(StandardOptions).streaming = True
+    p = TestPipeline(options=options)
+
+    main_stream = (p
+                   | 'main TestStream' >> TestStream()
+                   .advance_watermark_to(10)
+                   .add_elements(['e']))
+    side = (p
+            | beam.Create([2, 1, 4])
+            | beam.Map(lambda t: window.TimestampedValue(t, t)))
+
+    class RecordFn(beam.DoFn):
+      def process(self,
+                  elm=beam.DoFn.ElementParam,
+                  ts=beam.DoFn.TimestampParam,
+                  side=beam.DoFn.SideInputParam):
+        yield (elm, ts, side)
+
+    records = main_stream | beam.ParDo(RecordFn(), beam.pvalue.AsList(side)) | beam.Map(recorded_elements) # pylint: disable=line-too-long, unused-variable
```

Comment:
<!--new_thread; commit:2053e91ce5cf41b329a4bf8bdb48a06b79b3626b; resolved:0-->
Can you use the multi-line form to avoid the line-too-long?  You can also just avoid creating the `records` variable.

```
(main_stream
 | beam.ParDo(RecordFn, ...)
 | beam.Map(...))
```

Reply:
Done